src/engine/audio/audio.rs

```diff
@@ -0,0 +1,130 @@
+use std::collections::HashMap;
+use kira::{AudioManager, AudioManagerSettings, DefaultBackend, sound::static_sound::StaticSoundData, Tween};
+use std::path::Path;
+use nalgebra::Vector3;
+
+
+/// Stores a [StaticSoundData]
+pub struct AudioAsset {
+    sound_data: StaticSoundData,
+}
+
+/// Stores an [AudioManager] and a map of id's to audio assets
+pub struct AudioSystem {
+    manager: AudioManager<DefaultBackend>,
+    assets: HashMap<String, AudioAsset>,
+}
+
+/// Stores an [AudioSystem], the position of the receier (for now there is only one per scene),
+/// and the max distance that audio can be heard from the receiver
+pub struct AudioScene {
+    audio_system: AudioSystem,
+    receiver_position: Vector3<f32>,
+    max_distance: f32,
+}
+
+impl AudioScene {
+    /// Creates a new audio scene
+    pub fn new() -> Self {
+        Self {
+            audio_system: AudioSystem::new(),
+            receiver_position: Vector3::new(0.0, 0.0, 0.0),
+            max_distance: 50.0,
+        }
+    }
+
+    /// Loads a sound file from a path and saves it under id
+    pub fn load_sound(&mut self, id: &str, path: &str) {
+        self.audio_system.load_sound(id, path);
+    }
+
+    /// Sets the receiver position for the scene for spatial audio
+    pub fn set_receiver_position(&mut self, receiver_position: Vector3<f32>) {
+        self.receiver_position = receiver_position;
+    }
+
+    // Calculate linear volume scalar (0-1) based on distance between source and receiver and the max distance
+    pub fn calculate_volume_linear(distance: f32, max_distance: f32) -> f32 {
+        if distance >= max_distance {
+            0.0
+        } else {
+            (1.0 - (distance / max_distance)).clamp(0.0, 1.0)
+        }
+    }
+
+    // Converts normalized volume scalar to decibels
+    pub fn volume_to_db(volume: f32) -> f32 {
+        -60.0 + (0.0 - (-60.0)) * volume.clamp(0.0, 1.0)
+    }
+
+
+    /// Plays a sound from an id based off the position in the scene
+    pub fn play_sound(&mut self, sound_id: &str, position: Vector3<f32>) {
```

Comment:
Yea, this isn't exactly positional audio.

https://docs.rs/kira/latest/kira/track/index.html#spatial-tracks

Kira has spatial audio support, and you'll wanna add a listener in the Sound system aligned with creation of a SoundReceiver Component.